src/pages/workspaces/workspace/analysis/modals/ExportAnalysisModal/ExportAnalysisModal.component.ts

```diff
@@ -0,0 +1,133 @@
+import _ from 'lodash/fp'
+import { Fragment, useState } from 'react'
+import { b, h } from 'react-hyperscript-helpers'
+import { spinnerOverlay, useUniqueId } from 'src/components/common'
+import ErrorView from 'src/components/ErrorView'
+import Modal from 'src/components/Modal'
+import { WorkspaceSelector } from 'src/components/workspace-utils'
+import { AnalysisStateDependencies, analysisStateDependencies } from 'src/dependencies/analysis/analysis-state-resolver'
+import { ComponentDependencies, componentDependencies } from 'src/dependencies/component-resolver'
+import { Composable } from 'src/dependencies/dependency-core'
+import { FormLabel } from 'src/libs/forms'
+import { goToPath as navToPath } from 'src/libs/nav'
+import { isValidWsExportTarget, summarizeErrors } from 'src/libs/utils'
+import { WorkspaceInfo, WorkspaceWrapper } from 'src/libs/workspace-utils'
+import { getAnalysisFileExtension, stripExtension } from 'src/pages/workspaces/workspace/analysis/file-utils'
+import {
+  analysisNameInput, analysisNameValidator
+} from 'src/pages/workspaces/workspace/analysis/notebook-utils'
+import { analysisLauncherTabName, analysisTabName } from 'src/pages/workspaces/workspace/analysis/runtime-common-components'
+import { ToolLabel } from 'src/pages/workspaces/workspace/analysis/tool-utils'
+import validate from 'validate.js'
+
+
+export interface ExportAnalysisModalDeps {
+  components: Pick<ComponentDependencies, 'ButtonPrimary' | 'ButtonSecondary'>
+  analysisState: Pick<AnalysisStateDependencies, 'useAnalysisExportState'>
+}
+
+export interface ExportAnalysisModalProps {
+  fromLauncher?: boolean
+  onDismiss: (event: unknown) => void
+  printName: string
+  toolLabel: ToolLabel
+  workspace: WorkspaceWrapper
+}
+
+export type ComposableExportAnalysisModal = Composable<
+    ExportAnalysisModalDeps,
+    React.FC<ExportAnalysisModalProps>
+    >
+
+export const ExportAnalysisModal: ComposableExportAnalysisModal = {
+  compose: (deps: ExportAnalysisModalDeps) => {
+    const FC: React.FC<ExportAnalysisModalProps> = props => {
+      const { fromLauncher, onDismiss, printName, toolLabel, workspace: sourceWorkspace } = props
+      const { ButtonPrimary } = deps.components
+      const { useAnalysisExportState } = deps.analysisState
+
+      const uniqueId = useUniqueId()
```

Comment:
the rest of this FC code is from a pending IA team PR and not central to the Composable DI pattern being demonstrated here.  You can skim down to line 127